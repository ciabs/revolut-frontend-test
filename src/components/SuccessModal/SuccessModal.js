import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome  from 'react-fontawesome';

import {rateColor} from '../../styles/constants';
import {Container, Content} from './SuccessModal.styles';
import Button from '../Button/Button';
import {getCurrencySymbol} from '../../helpers/currency.helper';

const SuccessModal = ({handleHideModal, lastExchange}) => {

  const renderLastExchangeMessage = `You exchanged 
  ${getCurrencySymbol(lastExchange.fromCurrency)}${lastExchange.fromValue} â†’ 
  ${getCurrencySymbol(lastExchange.toCurrency)}${lastExchange.toValue}`;

  return (
    <Container>
      <Content>
        <div>
          <FontAwesome
            name='check-circle-o'
            size='5x'
            style={{
              color: rateColor,
              marginBottom: '30px'
            }}
          />
        </div>
        <span>
          {renderLastExchangeMessage}
        </span>
      </Content>
      <Button
        onClick={handleHideModal}
        isDisabled={false}
      >
        Done
      </Button>
    </Container>
  );
};

SuccessModal.propTypes = {
  handleHideModal: PropTypes.func.isRequired,
  lastExchange: PropTypes.object.isRequired
};

export default SuccessModal;
