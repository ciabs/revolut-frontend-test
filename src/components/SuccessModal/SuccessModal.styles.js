import styled from 'styled-components';
import {fromBackgroundColor} from '../../styles/constants';

export const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 5;
  width: 100%;
  height: 100%;
  background-color: ${fromBackgroundColor};
  display: flex;
  flex-direction: column;
  justify-content: center;
  border-radius: 10px;
`;

export const Content = styled.div`
  text-align: center;
  font-size: 18px;
`;
