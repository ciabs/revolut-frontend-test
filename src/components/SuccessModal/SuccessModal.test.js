import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SuccessModal from './SuccessModal';

describe('SuccessModal', () => {
  const props = {
    lastExchange: {},
    handleHideModal: jest.fn()
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <SuccessModal {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <SuccessModal {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
