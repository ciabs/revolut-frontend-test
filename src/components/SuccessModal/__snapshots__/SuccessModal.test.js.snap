// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa bdPopR"
>
  <div
    className="sc-bwzfXH fybyaO"
  >
    <div>
      <img
        className="sc-htpNat kmFITi"
        src="check.svg"
      />
    </div>
    <span>
      You exchanged 
        £undefined → 
        £undefined
    </span>
  </div>
  <button
    className="sc-bxivhb hNbXFO"
    disabled={false}
    onClick={[Function]}
  >
    Done
  </button>
</div>
`;
